### PR TITLE
feat(types): add support for relational sorting with dot notation

### LINKED ISSUE #24518: QuerySort missing relational type functionality in SDK
### Describe the Improvement

# Improve nested relational sort type safety

Currently, while the `QueryFields` type properly handles relational fields, the `QuerySort` type lacks proper type safety for nested relational sorting. This means TypeScript doesn't prevent potentially problematic sort operations like sorting by many-to-many fields (e.g., `m2m.id`).

Proposed improvement:
1. Add support for dot notation in sort paths to match the pattern used in `QueryFields`
2. Restrict sorting to only allow many-to-one relationships, preventing invalid sort attempts on many-to-many or many-to-any relations
3. Maintain TypeScript's ability to provide helpful error messages when invalid sort paths are used

Example of current vs desired behaviour:

```typescript
// Currently allowed but shouldn't be:
type Sort = ['m2m.id']  // many-to-many sorting shouldn't be allowed

// Should only allow many-to-one relations:
type ValidSort = ['m2o.id', 'string_field']  // ✓ valid
type InvalidSort = ['m2m.id']  // ✗ should fail type check
```

### PR DESCRIPTION
**Pull request by Jopgood**
# Add support for relational sorting with dot notation

## Scope
What's changed:
- Added support for sorting by related fields using dot notation (e.g. `m2o.id`)
- Updated `QuerySort` type to handle nested sort paths while maintaining type safety
- Added comprehensive test suite matching existing `QueryFields` test patterns
- Restricted sorting to only allow many-to-one relationships following existing codebase patterns

## Potential Risks / Drawbacks
- Edge cases in complex nested relationships might need additional testing
- Breaking change for users who might have been incorrectly using many-to-many sort paths?
- Performance impact for TypeScript type checking with more complex conditional types

## Review Notes / Questions
- Special attention should be paid to how we handle many-to-many relationships - currently blocked as unsortable following existing relationship patterns
- The implementation follows the same pattern as `QueryFieldsRelational` for consistency - would love feedback if this approach makes sense
- Test coverage matches the `QueryFields` test patterns - please let me know if additional test cases would be valuable

---
Fixes #24518

### CODE REVIEW

File: sdk/src/types/query.ts [hunk 1]
```diff
@@ -56,15 +56,25 @@ export type MergeFields<FieldList> = HasNestedFields<FieldList> extends never
...
 
+/**
+ * Handle relational field sorting using dot notation
+ */
+type RelationalSort<Schema, Item> = {
+	[Key in RelationalFields<Schema, Item>]: Extract<Item[Key], ItemType<Schema>> extends infer RelatedCollection
+		? RelatedCollection extends any[]
+			? HasManyToAnyRelation<RelatedCollection> extends never
+				? never // many-to-many or one-to-many should not be sortable
+				: never // many-to-any
```
Comment (br41nslug) [inline sdk/src/types/query.ts:73]: These are in fact sortable and should be represented here too.
image

**Review by br41nslug — CHANGES_REQUESTED** [review]
This relational sorting type seems to be limited to only 1 level of nesting however following relations can be infinitely recursive here.

After adding a m2o between collection_b and collection_a we should see infinite recursion:
image
image

Because of this recursive nature the dot notation doesnt lend itself well in typescript's type system for this exact reason we moved off from the dot notation for `fields` and are using an object based structure instead.

**Comment by AlexGaillard** [comment]
Hi @Jopgood, thanks for opening this PR. I'm going to close this out however as there's been no movement or response in some time. If you'd like to still try to fix the noted issue then please feel free to open a new PR. 😄
